Add methods and tests for tournament in storage pkg

HarlamovBuldog opened the pull request.

HarlamovBuldog committed:
Change field types in Tournament struct, improve error handling

Change type of users Array field from string array to pointer to string array.
This will cause mongodb to init array inside itself.
Because otherwise string array will be nil and cause errors.
Also we need to init this array when creating new Tournament.
Improved error handling in AddUserToTournamentList and SetTournamentWinner funcs.
Because we need to check both tournament and user id here.
So errors was made more informational to understand what actually cause them.

HarlamovBuldog committed:
Impement 2 methods and add 1 new method for tournament in storage package.

HarlamovBuldog committed:
Add tests for tournament methods inside storage package.

pisarukv commented on internal/storage/tournament.go:
```diff
@@ -18,8 +18,8 @@ type Tournament struct {
 	Prize   float64            `json:"prize" bson:"prize"`
-	Users   []string           `json:"users" bson:"users"`
-	Winner  *User              `json:"winner" bson:"winner"`
+	Users   *[]string          `json:"users" bson:"users"`
```
why pointer

pisarukv commented on internal/storage/tournament.go:
```diff
@@ -84,20 +85,21 @@ func (db *DB) DeleteTournament(ctx context.Context, id string) error {
 	return nil
 }
 
+// AddUserToTournamentList func ..
```
.. ??

pisarukv commented on internal/storage/tournament.go:
```diff
@@ -113,20 +115,21 @@ func (db *DB) AddUserToTournamentList(ctx context.Context, tournamentID, userID
 	}
 
-	primUserID, err := primitive.ObjectIDFromHex(userID)
+	_, err = primitive.ObjectIDFromHex(userID)
```
why not to use object id

HarlamovBuldog committed:
Change type of fields Users and Winner for Tournament in storage pkg.

Change type of fields Users and Winner from string to primitive.ObjectID.
Also remove pointer from Users array field as unnecessary.

HarlamovBuldog committed:
Add test for SetTournamentStatus method.

Correct error wrap inside SetTournamentStatus method to be more informative.

HarlamovBuldog committed:
Add missing methods comments in storage pkg.
